vhost-device-scmi/src/devices/iio.rs

```diff
@@ -396,7 +396,9 @@ impl IIOSensor {
         let os_file_name = file.file_name();
         let file_name = os_file_name.to_str().unwrap_or_default();
         let raw_suffix = "_raw";
-        if file_name == "name" {
+        // Only override name when no defined by parameter.
+        // If both fs and parameter exist, use parameter first and skip from file.
+        if file_name == "name" && self.sensor.name == "iio" {
```

Comment:
Hm, if somebody gets an idea to use "iio" as the name on the command line, this won't work and will cause confusion. Changing the default name to something less likely to be used (something like "_iio_default" or "(IIO)") would be the easiest fix. Another option is to track whether the default name was used but it wouldn't be very elegant.

Reply:
Because there is a default name "iio", then if someboay want to use "iio" as the name on command line. It will still be "iio".
But your idea is good, it actually confuse developer to read this code here.
Should we make a macro to define default name?

For example, at the beginning of the file, define a macro like below
```
diff --git a/vhost-device-scmi/src/devices/iio.rs b/vhost-device-scmi/src/devices/iio.rs
index 97d23a4..43b88f4 100644
--- a/vhost-device-scmi/src/devices/iio.rs
+++ b/vhost-device-scmi/src/devices/iio.rs
@@ -163,6 +163,8 @@ const UNIT_MAPPING: &[UnitMapping] = &[
     },
 ];

+const IIO_DEFAULT_NAME: &str = "iio";
+
 /// Representation of an IIO channel axis.
 ///
 /// Used also for scalar values.
@@ -433,7 +435,7 @@ impl IIOSensor {
     #[allow(clippy::new_ret_no_self)]
     pub fn new(properties: &DeviceProperties) -> Result<Self, DeviceError> {
         properties.check(&["path", "channel"], &["name"])?;
-        let sensor = Sensor::new(properties, "iio");
+        let sensor = Sensor::new(properties, IIO_DEFAULT_NAME);
         Ok(Self {
             sensor,
             path: OsString::from(properties.get("path").unwrap()),
@@ -502,7 +504,7 @@ impl IIOSensor {
         let raw_suffix = "_raw";
         // Only override name when no defined by parameter.
         // If both fs and parameter exist, use parameter first and skip from file.
-        if file_name == "name" && self.sensor.name == "iio" {
+        if file_name == "name" && self.sensor.name == IIO_DEFAULT_NAME {
             self.set_sensor_name_from_file(&file.path());
         } else if file_name.starts_with(channel) && file_name.ends_with(raw_suffix) {
```

It seems better to understand this logic, do you think so?

Reply:
> Because there is a default name "iio", then if someboay want to use "iio" as the name on command line. It will still be "iio". 

It won't because the name will be read from the file system in such a case.

> It seems better to understand this logic, do you think so?

Yes, it is in any case.

Reply:
All right.